target/portfolio-1/js/createGame.js

```diff
@@ -0,0 +1,53 @@
+function addNewStage() {
+  const activeStages = document.getElementsByClassName('activeStage')[0];
+  activeStages.classList.remove('activeStage');
+  const divToHide = activeStages.classList[0];
+  document.getElementById(divToHide + 'Hints').classList.add('hidden');
+  const stagesList = document.getElementById('stages');
+  const numStages = stagesList.getElementsByTagName('input').length;
+
+  const newStage = document.createElement('input');
+  newStage.value = 'Stage ' + (numStages + 1);
+  newStage.type= 'button';
+  newStage.className = 'stage' + (numStages + 1) + ' activeStage';
+
+  newStage.addEventListener("click", function(){
+    setActive('stage' + (numStages + 1));
+  });
+
+  const a = document.createElement('a');
+  a.href = '#stage' + (numStages + 1);
+
+  a.appendChild(newStage);
+  stagesList.appendChild(a);
+
+  const newStageHints = document.createElement('div');
+  newStageHints.id = newStage.classList[0] + 'Hints';
+  newStageHints.innerHTML = "<input type='text' placeholder='Starter (spawn location and first hint)' id='starter-hint' required>" +
+                            "<input type='text' placeholder='Hint 1' id='hint1'>";
+
+  document.getElementById('hints').appendChild(newStageHints);
+}
+
+function addNewHint() {
+  const activeStageNum = (document.getElementsByClassName('activeStage')[0]).classList[0] + 'Hints';
```

Comment:
Might want to write helper functions getActiveStageElement() and getStageNumber(stageElement) to make stuff like this clearer.

Reply:
Okay, sounds good!